### PR TITLE
Add CI Validation

### PR DESCRIPTION
**Pull request by orionw**
- Adds CI validation for pipelines, primitive jsons

### CODE REVIEW

File: submission/ci_validation.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+primitive_jsons = Path(byu_dir).glob('**/primitive.json')
+# for filename in primitive_jsons:
+#     # strip the init file path off, since we are changing directories
+#     filename_real = "/".join(str(filename).split("/")[2:])
+#     print("Validating primitive:", filename_real)
+#     os.system("cd submission/primitives/; python3 run_validation.py {}; cd ../..".format(filename_real))  
+
+# check pipelines and meta files
+for filename in Path(byu_dir).glob('**/*.json'):
+    if filename in primitive_jsons:
```
Comment (bjschoenfeld) [inline submission/ci_validation.py]: I wonder if this is not working

Comment (orionw) [reply submission/ci_validation.py]: Apparently glob uses a PosixPath and so it isn't a string. That should be fixed now

File: submission/pipelines/generate_pipelines.py [hunk 2]
```diff
@@ -543,7 +546,8 @@ def main():
         write_pipeline_for_submission(os.path.join(byu_dir, __metafeature_path__), str(__metafeature_version__), pipeline_json_structure, problem_name)
 
     # add other best pipelines
-    add_best_pipelines(byu_dir)
+    # TODO: update the experimenter to produce valid pipelines
+    # add_best_pipelines(byu_dir)
```
Comment (bjschoenfeld) [inline submission/pipelines/generate_pipelines.py]: I know there is a reason for this, but I can't remember... Can you remind me?

Comment (orionw) [reply submission/pipelines/generate_pipelines.py]: Yes, once we update the experimenter and can get valid pipelines.  Right now we have to manually update them after we grab them

**Pull request merged**